src/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,157 @@
+import {
+	Component,
+	Input,
+	Output,
+	ViewChild,
+	EventEmitter,
+	OnInit
+} from "@angular/core";
+import { NG_VALUE_ACCESSOR } from "@angular/forms";
+
+import { I18n } from "../i18n/i18n.module";
+import { FileItem } from "./file-uploader.interface";
+
+const noop = () => {
+};
+
+@Component({
+	selector: "ibm-file-uploader",
+	template: `
+		<strong class="bx--label">{{labelTitle}}</strong>
+		<p class="bx--label-description">{{labelDescription}}</p>
+		<div class="bx--file">
+			<button
+				ibmButton="secondary"
+				(click)="file.click()"
+				[attr.for]="fileUploaderId">
+				{{buttonLabel}}
+			</button>
+			<input
+				#file
+				type="file"
+				class="bx--file-input"
+				[accept]="accept"
+				[id]="fileUploaderId"
+				[multiple]="multiple"
+				(change)="onFilesAdded()"/>
+			<div class="bx--file-container">
+				<ibm-file *ngFor="let content of files"
+					[content]="content"
+					[files]="files"
+					[file]="file"
+					[filesChange]="filesChange">
```

Comment:
And individual file shouldn't need to know this much to render ... you should be able to just pass in a `FileItem` and listen to a `remove` event and have `FileUploader` handle adding/removing and notifying the consumer